lib/main-window.js

```diff
@@ -71,6 +73,44 @@ module.exports = function (config) {
         api.profile.sheet.edit({ usePreview: false })
       }
     })
+
+    const currentlyDeleting = {}
+
+    watch(api.contact.obs.blocking(id), (blocking) => {
+      if (blocking.length === 0) return
+
+      if (ssb.del == null) return
+
+      blocking.forEach(feed => {
+        pull(
+          ssb.createUserStream({ id: feed }),
+          pull.asyncMap((msg, cb) => {
+            const key = msg.key
+            if (currentlyDeleting[key] === true) {
+              return cb(null, null) // already deleting
```

Comment:
Thanks for taking a look!

The deletion kicks off each time the blocklist changes, so this can happen if the database is indexing and finding new people you've blocked *or* if the user is rapidly blocking people.

I think the race condition here is fine since it's okay to delete the same message multiple times, it's just inefficient so it's nice to be able to reduce the repeat deletes when we can.

Reply:
Oh, and I guess this part is sync so it wouldn't be a race condition anyway 😁